authors/apps/profiles/tests/test_follows.py

```diff
@@ -0,0 +1,102 @@
+# from authors.apps.authentication.messages import error_msg, success_msg
+
+class UserFollow(APITestCase):
+    """This class tests for user follow and unfollow"""
+    def setUp(self):
+        self.register_url = reverse("auth:register")
+        self.login_uri = reverse('auth:login')
+        self.valid_user_credentials = {
+            "user": {
+                "username":"wachira",
+                "email":"ewachira254@gmail.com",
+                "password":"Wachira254"
+            }
+        }
+        self.valid_user2_credentials = {
+            "user": {
+                "username":"kerubo",
+                "email":"kerubo@gmail.com",
+                "password":"Wachira254"
+            }
+        }
+    def user_login(self):
+        self.client.post(
+                self.register_url,
+                self.valid_user_credentials,
+                    format="json"
+        )
+        response = self.client.post(
+            self.login_uri,
+            data=self.valid_user_credentials,
+            format="json"
+        ) 
+        token = response.data['token']
+        return token
+    def user2_login(self):
+        self.client.post(
+                self.register_url,
+                self.valid_user2_credentials,
+                format="json"
+        )
+        response = self.client.post(
+            self.login_uri,
+            data=self.valid_user2_credentials,
+            format="json"
+        ) 
+        token = response.data['token']
+        return token
+
+    def follow(self):
+        self.user_login()
+        token = self.user2_login()
+        follow_url = reverse('profile:follow', kwargs={'username': 'wachira'})
+        response = self.client.post(follow_url, HTTP_AUTHORIZATION='Bearer ' + token, format='json')
+        return response
+    
+    def test_follow(self):
+        response = self.follow()
+        # import pdb; pdb.set_trace()
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertIn(response.data[1]['message'],
+                      success_msg['success_followed'])
+
+    def test_unfollow(self):
+        self.follow()
+        token = self.user2_login()
+        unfollow_url = reverse('profile:follow', kwargs={'username': 'wachira'})
+        response = self.client.delete(unfollow_url, HTTP_AUTHORIZATION='Bearer ' + token, format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIn(response.data[1]['message'],
+                      success_msg['success_unfollowed'])
+
+    def test_follow_self(self):
+        token  = self.user_login()
+        follow_url = reverse('profile:follow', kwargs={'username': 'wachira'})
+        response = self.client.post(follow_url, HTTP_AUTHORIZATION='Bearer ' + token, format='json')
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertIn(response.data, "You cannot follow yourself")
```

Comment:
@Deekerubo use the messages dictionary as opposed to hardcoding the strings.

Reply:
Done